src/cryptonote_core/blockchain.cpp

```diff
@@ -3477,6 +3479,7 @@ bool Blockchain::cleanup_handle_incoming_blocks(bool force_sync)
   CRITICAL_REGION_LOCAL(m_blockchain_lock);
   TIME_MEASURE_START(t1);
 
+  m_db->batch_stop();
```

Comment:
This does it unconditionally, whatever the call below returns ? If there is none active, an exceptoiun will be thrown, so we'll miss the rest of the cleanup